### PR TITLE
fix: lots of corner cases in PromQL

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Changes 😇
- don't generate filter plan if there is no filter
- don't include timestamp filter in filter plan
- fix how the offset is handled
- consider the stale NaN case
- don't filter NaN for instant selector
- ignore nonexistent time series in range selector

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
#1042

### CODE REVIEW

**Review by fengjiachun — APPROVED** [review]
lgtm

File: src/promql/src/extension_plan/instant_manipulate.rs [hunk 1]
```diff
@@ -49,6 +49,8 @@ pub struct InstantManipulate {
     lookback_delta: Millisecond,
     interval: Millisecond,
     time_index_column: String,
+    /// A optional column for validating staleness
+    value_column: Option<String>,
```
Comment (killme2008) [inline src/promql/src/extension_plan/instant_manipulate.rs:53]: is it `field_column`?

**Review by yuanbohan — APPROVED** [review]

File: src/promql/src/extension_plan/instant_manipulate.rs (hunk 1 above)
Comment (waynexia) [reply src/promql/src/extension_plan/instant_manipulate.rs]: Forget this... 🫣

File: src/promql/src/extension_plan/instant_manipulate.rs [hunk 2]
```diff
@@ -333,7 +333,7 @@ impl InstantManipulateStream {
             // then, search backward to lookback
             loop {
                 let curr = ts_column.value(cursor);
-                if let Some(value_column) = &value_column && value_column.value(cursor).is_nan() {
+                if let Some(value_column) = &field_column && value_column.value(cursor).is_nan() {
```
Comment (killme2008) [inline src/promql/src/extension_plan/instant_manipulate.rs:336]: Some(field_column)

**Pull request merged**